prime-router/src/main/kotlin/history/azure/SubmissionFunction.kt

```diff
@@ -26,14 +27,21 @@ class SubmissionFunction(
     submissionsFacade,
     workflowEngine
 ) {
+    /**
+     * We store the service name here to pass to the facade
+     */
+    var sendingOrgSvc: String? = null
+
     /**
      * Get the correct name for an organization sender based on the name.
      *
      * @param organization Name of organization and service
      * @return Name for the organization
      */
     override fun getOrgName(organization: String): String? {
-        return workflowEngine.settings.findSender(organization)?.organizationName
+        return workflowEngine.settings.findSender(organization).also {
+            if (organization.contains(Sender.fullNameSeparator)) sendingOrgSvc = it?.name
+        }?.organizationName
```

Comment:
If a fully qualified org.sender string was supplied, store the validated sender name to use during our query later on.